### PR TITLE
Convert Holonomic Functions to Hypergeometric and eventually to expressions/functions.

### PR DESCRIPTION
**Pull request by shubhamtibra**

### CODE REVIEW

**Comment by shubhamtibra** [comment]
I have written the method for converting to `hyper` when the order of recurrence is `One`. I can't find examples to test this on except `exp(x)`. Ping @jksuom @certik .

**Comment by jksuom** [comment]
`(1 + x)**a` would be an example (though fairly trivial).

**Comment by certik** [comment]
Great job! I'll try to play with everything that you have done in the coming days and we can discuss what to do in the second half of the GSoC. Great progress overall.

File: sympy/holonomic/holonomic.py [hunk 1]
```diff
@@ -1228,6 +1202,200 @@ def _find_conditions(simp, x, x0, order, evalf=False):
...
+            for i, j in enumerate(den):
+                den[i] = j.p
+
+        denom.append(R(den))
+
+    # lcm of denominators in the coefficients
+    for i in denom:
+        lcm_denom = i.lcm(lcm_denom)
+
+    if negative is True:
```
Comment (debugger22) [inline sympy/holonomic/holonomic.py]: Maybe just `if negative:` will work fine.

File: sympy/holonomic/holonomic.py [hunk 2]
```diff
@@ -1228,6 +1202,200 @@ def _find_conditions(simp, x, x0, order, evalf=False):
...
+    sol.append(DMFdiff(listofpoly[0]))
+
+    for i, j in enumerate(listofpoly[1:]):
+        sol.append(DMFdiff(j) + listofpoly[i])
+
+    sol.append(listofpoly[a])
+    return sol
+
+
+def _add_lists(list1, list2):
```
Comment (debugger22) [inline sympy/holonomic/holonomic.py]: The name `_add_lists` doesn't really tell what is actually does. A docstring explaining the function would be good.

Comment (shubhamtibra) [inline sympy/holonomic/holonomic.py]: Thanks @debugger22 for your reviews. I will make the appropriate changes.

**Comment by shubhamtibra** [comment]
I have added the general case method to convert Holonomic Function to linear combination of Hypergeometric. Ping @jksuom @certik .

**Comment by shubhamtibra** [comment]
@jksuom @certik I have implemented the algorithm, can you please take a look at it and share your views?

**Comment by shubhamtibra** [comment]
I modified the method `.to_sequence()`, which now also returns a value `n0` so that, the recurrence holds for all `n>n0`. This solved bugs in converting to `Hypergeometric`.

**Comment by shubhamtibra** [comment]
Ping @certik @jksuom .

**Comment by certik** [comment]
@shubhamtibra I think this looks great. Merging.

**Pull request merged**